File: src/utils/clipboard.ts
import { invoke } from "@tauri-apps/api/core";

/**
 * Copy text to the system clipboard via the Rust side.
 *
 * `navigator.clipboard.writeText` is denied by Windows WebView2 (no host
 * permission), so every copy button routes through this instead. Throws on
 * failure so callers can show their own feedback.
 */
export async function copyText(text: string): Promise<void> {
  await invoke("copy_text", { text });
}

File: src-tauri/src/clipboard.rs
//! One reliable copy-to-clipboard path for the frontend.
//!
//! `navigator.clipboard.writeText` is denied by Windows WebView2 unless the
//! host grants the clipboard permission, so buttons built on it silently did
//! nothing there (first reported: the offline-models system-info copy).
//! Copying on the Rust side talks to the OS clipboard directly and works the
//! same on every platform.

use tauri::AppHandle;

#[tauri::command]
pub fn copy_text(app: AppHandle, text: String) -> Result<(), String> {
    use tauri_plugin_clipboard_manager::ClipboardExt;
    app.clipboard()
        .write_text(text)
        .map_err(|e| format!("Could not copy to the clipboard: {e}"))
}
